pyiron_atomistics/atomistics/structure/neighbors.py

```diff
@@ -363,28 +394,25 @@ def _get_vectors(
             if distances is None or indices is None:
                 distances, indices = self._get_distances_and_indices(
                     positions=positions,
-                    allow_ragged=False,
+                    mode='filled',
                     num_neighbors=num_neighbors,
                     cutoff_radius=cutoff_radius,
                     width_buffer=width_buffer,
                 )
             vectors = np.zeros(distances.shape+(3,))
-            vectors -= self._get_wrapped_positions(positions).reshape(distances.shape[:-1]+(-1, 3))
+            vectors -= self._get_wrapped_positions(
+                positions
+            ).reshape(distances.shape[:-1]+(-1, 3))
             vectors[distances<np.inf] += self._get_extended_positions()[
                 self._extended_indices[distances<np.inf]
             ]
             vectors[distances==np.inf] = np.array(3*[np.inf])
-        elif self.vecs is not None:
-            vectors = self.vecs
         else:
-            raise AssertionError(
-                'vectors not created yet -> put positions or reinitialize with t_vec=True'
-            )
-        if allow_ragged==self.allow_ragged:
-            return vectors
-        if allow_ragged:
-            return self._contract(vectors, distances)
-        return self._fill(vectors)
+            if self._vecs is None:
```

Comment:
I don't understand, what is the point if assigning a variable to `_` and then never grabbing `_` later? Is this some sort of garbage collecting trick? As far as I can see these two lines can be removed and the functionality is the same, but if I'm wrong I will learn something new 😄

Reply:
That's because inside `def vecs(self)`, `self._vecs` is initialized (no idea if this is a particularly elegant way to do that). Anyway following last week's pyiron meeting this functionality is going to be removed.